server/schedule/region_scatterer.go

```diff
@@ -127,24 +127,30 @@ type RegionScatterer struct {
 // RegionScatter is used for the `Lightning`, it will scatter the specified regions before import data.
 func NewRegionScatterer(ctx context.Context, cluster Cluster) *RegionScatterer {
 	return &RegionScatterer{
-		ctx:            ctx,
-		name:           regionScatterName,
-		cluster:        cluster,
-		ordinaryEngine: newEngineContext(ctx, filter.NewEngineFilter(regionScatterName, filter.NotSpecialEngines)),
+		ctx:     ctx,
+		name:    regionScatterName,
+		cluster: cluster,
+		ordinaryEngine: newEngineContext(ctx, func() filter.Filter {
+			return filter.NewEngineFilter(regionScatterName, filter.NotSpecialEngines)
+		}),
 		specialEngines: make(map[string]engineContext),
 	}
 }
 
+type filterFunc func() filter.Filter
+
 type engineContext struct {
-	filters        []filter.Filter
+	filterFuncs    []filterFunc
```

Comment:
Do we need to it in `engineContext`?

Reply:
This makes filters easy to use, I think we can keep it.